src/commands/typescript/update.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { fs, Messages, SfdxError } from '@salesforce/core';
+import { exec } from 'shelljs';
+import { set } from '@salesforce/kit';
+import { asObject, ensureString, get } from '@salesforce/ts-types';
+import { NpmPackage, Package } from '../../package';
+import { SinglePackageRepo } from '../../repository';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-release-management', 'typescript.update');
+
+export default class Update extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly flagsConfig: FlagsConfig = {
+    version: flags.string({
+      char: 'v',
+      description: messages.getMessage('typescriptVersion'),
+    }),
+    target: flags.string({
+      char: 't',
+      description: messages.getMessage('esTarget'),
+      default: 'ESNext',
+    }),
+  };
+
+  public async run(): Promise<void> {
+    this.validateEsTarget();
+    const typescriptPkg = this.retrieveTsPackage();
+    this.validateTsVersion(typescriptPkg);
+
+    await this.updateTsVesion(typescriptPkg);
+    await this.updateEsTarget();
+
+    const pkg = await SinglePackageRepo.create(this.ux);
+    pkg.install();
+    pkg.build();
+    pkg.test();
+  }
+
+  private async updateEsTarget(): Promise<void> {
+    const tsConfigPath = path.resolve('tsconfig.json');
+    const tsConfig = await fs.readJson(tsConfigPath);
+    set(asObject(tsConfig), 'compilerOptions.target', this.flags.target);
+    this.ux.log(`Updating tsconfig target at ${tsConfigPath} to:`, this.flags.target);
+    await fs.writeJson(tsConfigPath, tsConfig);
+  }
+
+  private async updateTsVesion(typescriptPkg: NpmPackage): Promise<void> {
+    const newVersion = ensureString(this.flags.version || get(typescriptPkg, 'dist-tags.latest'));
+    this.ux.log(`Updating typescript version to ${newVersion}`);
+    const pkg = await Package.create(path.resolve('.'));
+
+    if (pkg.packageJson.devDependencies['typescript']) {
+      pkg.packageJson.devDependencies['typescript'] = newVersion;
+    }
+
+    // If the prepare script runs sf-install, the install will fail because the typescript version
+    // won't match the expected version. So in that case, we delete the prepare script so that we
+    // get a successful install
+    if (pkg.packageJson.scripts['prepare'] === 'sf-install') {
+      delete pkg.packageJson.scripts['prepare'];
+    }
```

Comment:
Do we ever need to put this back into the package?

Reply:
@WillieRuemmele No, these changes aren't being committed - we just need them so we can run the tests in circle with the updated ts version

Reply:
What about running these locally? I think we should make it clear (output to the use at the end of the run) that any file changes from this command should NOT be committed.

Reply:
i.e. This is for testing new versions only and to actually update the version it must go through dev-scripts.